### ISSUE TITLE
[Feature]: Date Comparison transform for Assertions

### ISSUE THREAD

**Issue by SuddenDevelopment** [apisec-inc/Feature-Request#51]
### Please describe your problem?

Cert variables are in QA, but the dates are an ISO string which we cannot compare

### How important is this feature?

Very Important

### Describe the solution you'd like?

add a transform for the ISO date time string to unixtime in seconds

Important because it's required to do categories for the Cert vars

**Comment by jaleelsyed** [comment]
Comprison operators for date are dateEquals, dateBefore, and dateAfter.
Ex. 
'@Response.data.createdDate dateEquals @Response.data.createdDate'
'@Response.data.createdDate dateBefore  2022-10-20'
'@Response.data.createdDate dateAfter  2022-10-20'

**Comment by SuddenDevelopment** [comment]
@jaleelsyed  can you please show me an example that will compare against current time?

I need to see if a certificate expires in 30 days, or if a certificate expires tomorrow, or if it's already expired.

**Comment by intesar** [comment]
@Mohammed-Luqman-Shareef 

Are these variables supported?
'@Response.data.createdDate dateEquals @Response.data.createdDate'
'@Response.data.createdDate dateBefore 2022-10-20'
'@Response.data.createdDate dateAfter 2022-10-20'

Can you provide an example with @Cert-ExpiresOn

**Comment by Mohammed-Luqman-Shareef** [comment]
@intesar,  Yes those three date comparison variables **dateEquals**, **dateAfter**, **dateBefore** are supported for various commonly used date and datetime formats.

Below are the working examples

- '@Cert-ExpiresOn dateEquals 2022-10-20'
- '@Cert-ExpiresOn dateAfter 2022-10-20'
- '@Cert-ExpiresOn dateBefore 2022-10-20'
- '@Cert-ExpiresOn dateAfter @CurrentDate'
- '@Response..data.createdDate dateEquals @Response..data.createdDate'

**Comment by SuddenDevelopment** [comment]
I don't see how that will meet the requirement mentioned of being able to do the date math. 

I need to do an assertion if it will expire tomorrow for example. 

How do I add days to the current date variable?

**Comment by jaleelsyed** [comment]
Try this way **@Cert-ExpiresOn** daterAfter **2022-11-15**. or you may also try any other valid date format.

**Comment by SuddenDevelopment** [comment]
@jaleelsyed  No, this is the exact same issue we've discussed a few times now. If we stay with string dates I cannot do date math. I cannot do an assertion that looks 1 day or 1 week ahead I have to use exact dates, pleas re-read the previous comment and create a solution to do the date math or dat transform

**Comment by Mohammed-Luqman-Shareef** [comment]
@jaleelsyed 
As discussed over the call, create 4 new vairables to return the future dates
@NextDay
@NextWeek
@NextMonth
@NextYear

**Comment by Mohammed-Luqman-Shareef** [comment]
A quick question:
   Since we hit customer's Test environment and not their production, does this cert expiry check in test env makes sense? and will it be considered an actual vulnerability?

**Comment by jaleelsyed** [comment]
New variables were added as 

- '@NextDay'
- '@NextWeek'
- '@NextMonth'
- '@NextYear'

Eg: - '@Response.data.createdDate dateEquals @NextDay'

**Comment by SuddenDevelopment** [comment]
no matter how the cert thing is used the future dates can be useful on other things like cookies

**Comment by SuddenDevelopment** [comment]
@jaleelsyed 

please normalize the caps in variables. choose all lowercase or camelcase. Month is different than the rest

**Comment by SuddenDevelopment** [comment]
it doesnt look like @nextweek is implemented in prod. I get null

**Comment by jaleelsyed** [comment]
Its in camelcase only, for some reason its taking in lowercase.

**Comment by SuddenDevelopment** [comment]
@jaleelsyed  I haven't found a version that appears to work in prod:
[@Cert-ExipresOn < @nextWeek] resolved-to [null < null]
[@Cert-ExipresOn < @NextWeek] resolved-to [null < null] result [Failed]

**Comment by jaleelsyed** [comment]
@SyedsImran Pls refresh the prod scanners.

**Comment by SyedsImran** [comment]
Refreshed prod scanners.

**Comment by SuddenDevelopment** [comment]
confirmed to work in prod, thanks
